catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreAutoPublisher.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.api.impl;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.codice.ddf.registry.api.RegistryStore;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.codice.ddf.security.common.Security;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.source.SourceMonitor;
+
+public class RegistryStoreAutoPublisher implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryStoreAutoPublisher.class);
+
+    private FederationAdminService federationAdminService;
+
+    private RegistryPublicationService registryPublicationService;
+
+    private ConcurrentHashMap<String, Boolean> registryStoreMap = new ConcurrentHashMap<>();
+
+    public void setFederationAdminService(FederationAdminService federationAdminService) {
+        this.federationAdminService = federationAdminService;
+    }
+
+    public void setRegistryPublicationService(
+            RegistryPublicationService registryPublicationService) {
+        this.registryPublicationService = registryPublicationService;
+    }
+
+    public void bindRegistryStore(ServiceReference reference) {
```

Comment:
Good Call. I thought I could get away without one, but then I started using the Map to track registries I had seen. Unbind clears the unbound registry out of the map now.